libpod/healthcheck.go

```diff
@@ -407,18 +407,21 @@ func (c *Container) getHealthCheckLog() (define.HealthCheckResults, error) {
 	return healthCheck, nil
 }
 
-// HealthCheckStatus returns the current state of a container with a healthcheck
+// HealthCheckStatus returns the current state of a container with a healthcheck.
+// Returns an empty string if no health check is defined for the container.
 func (c *Container) HealthCheckStatus() (string, error) {
-	c.lock.Lock()
-	defer c.lock.Unlock()
+	if !c.batched {
+		c.lock.Lock()
+		defer c.lock.Unlock()
+	}
 	return c.healthCheckStatus()
 }
 
 // Internal function to return the current state of a container with a healthcheck.
 // This function does not lock the container.
 func (c *Container) healthCheckStatus() (string, error) {
 	if !c.HasHealthCheck() {
-		return "", fmt.Errorf("container %s has no defined healthcheck", c.ID())
+		return "", nil
 	}
 
 	if err := c.syncContainer(); err != nil {
```

Comment:
I think the syncContainer() here should be moved in the `if !batched` branch just like the other public accessor functions

Reply:
I am sure there's more to polish in the health-check code but it's out of scope of this PR.